### ISSUE TITLE
URL rewrites are not generated when editing or duplicating a product

### ISSUE THREAD

**Issue by densen45** [magento/magento2#33884]
This issue occurs only when the Single-Store mode is enabled. It holds for editing existing products and creating duplicates as well. Below only the detailed steps for editing an existing product are provided to reproduce this issue. However, if you want to reproduce the issue for duplicating products, just duplicate an existing product and you will see, that no URL will be generated for the duplicate.

### Preconditions (*)

1. 2.4.2 or 2.4.2-p2

### Steps to reproduce (*)

1. Activate the Single-Store mode
2. Create a product and add it to a category 
2. Go to the URL Rewrites page (Marketing > SEO & Search > URL Rewrites) and notice the generated URLs.
4. Change the URL Key of the product and save it
6. Go to the URL Rewrites page again

### Expected result (*)

1. URLs for the product were changed (like <category_key>/<product_key>.html and <product_key>.html).
2. When you open the category on the frontend and hover over the product a correct URL (from above) should be previewed and opened by click.
3. When you use the URL key to directly access the product on frontend like https://<the_domain>/<url_key>.html the page is found.

### Actual result (*)

1. No URL for the product were changed (no changed entry can be found on the URLs Rewrite page and in the DB (url_rewrite table) as well).
2. When you open the category on the frontend and hover over the product an unaesthetic URL gets previewed (like https://<the_domain>/catalog/product/view/id/<product_id>/s/<product_url_key>/<category_id>/)
3. When you use the URL key to directly access the product on frontend like https://<the_domain>/<url_key>.html the page is not found.

### Probably related issues

* #25190
* #30316

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by densen45** [comment]
@magento give me 2.4-develop instance

**Comment by densen45** [comment]
I cannot reproduce the issue on 2.4-develop instance.

**Comment by densen45** [comment]
@magento give me 2.4.2 instance

**Comment by densen45** [comment]
I'm closing this issue because I cannot reproduce it on 2.4-develop and 2.4.2 instances.

**Comment by densen45** [comment]
@magento give me 2.4.2 instance

**Comment by densen45** [comment]
I have now been able to reproduce the problem on a 2.4.2 instance. The key was to enable the Single-Store mode.

**Comment by densen45** [comment]
@magento give me 2.4-develop instance

**Comment by densen45** [comment]
I'm still not able to reproduce this issue on a 2.4-develop instance. Enabling Single-Store mode does not cause this problem. However, on a 2.4.2 instance, the problem still occurs.

**Comment by densen45** [comment]
I have found the issue #25190 which provides the answer and the solution for this issue here in https://github.com/magento/magento2/issues/25190#issuecomment-593350246.

I have checked the current 2.4-develop code and the three lines mentioned in the linked comment above were removed in this commit: 494eb40eb236b9874fba78588fb3bf86b6525d65
Thus, I assume that removing these lines will not affect anything else.

So I have created a module that overrides the `getStoreIds()` function of the `\Magento\Catalog\Model\Product` class and got rid of the erroneous code. This solves the issue for editing and duplicating products as well.

I'm closing this issue because the fix was already released in 2.4.3.
